app/presenters/publish_warning_presenter.rb

```diff
@@ -0,0 +1,73 @@
+class PublishWarningPresenter
+  attr_reader :service, :action
+
+  MESSAGE = {
+    publish: {
+      both_pages: I18n.t('publish.warning.both_pages'),
+      cya_page: I18n.t('publish.warning.cya'),
+      confirmation_page: I18n.t('publish.warning.confirmation')
+    },
+    delete: {
+      both_pages: I18n.t('pages.flow.delete_warning_both_pages'),
+      cya_page: I18n.t('pages.flow.delete_warning_cya_page'),
+      confirmation_page: I18n.t('pages.flow.delete_warning_confirmation_page')
+    }
+  }.freeze
+
+  def initialize(service, action)
+    @service = service
+    @action = action
+  end
+
+  def message
+    submitting_pages_not_present_message ||
+      confirmation_page_not_present_message ||
+      cya_page_not_present_message
+  end
```

Comment:
since this method is being called twice you can memoize the result so that the logic only has to run once.

also i think everything below this method can be `private`